src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/PartialTransactionRequester.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.Utilities;
+using Stratis.FederatedPeg.Features.FederationGateway.Interfaces;
+using Stratis.FederatedPeg.Features.FederationGateway.NetworkHelpers;
+
+namespace Stratis.FederatedPeg.Features.FederationGateway.TargetChain
+{
+    public class PartialTransactionRequester : IPartialTransactionRequester
+    {
+        private readonly ILoggerFactory loggerFactory;
+        private readonly ILogger logger;
+        private readonly ICrossChainTransferStore crossChainTransferStore;
+        private readonly IAsyncLoopFactory asyncLoopFactory;
+        private readonly INodeLifetime nodeLifetime;
+        private readonly IConnectionManager connectionManager;
+        private readonly IFederationGatewaySettings federationGatewaySettings;
+
+        private IAsyncLoop asyncLoop;
+
+        public PartialTransactionRequester(
+            ILoggerFactory loggerFactory,
+            ICrossChainTransferStore crossChainTransferStore,
+            IAsyncLoopFactory asyncLoopFactory,
+            INodeLifetime nodeLifetime,
+            IConnectionManager connectionManager,
+            IFederationGatewaySettings federationGatewaySettings)
+        {
+            Guard.NotNull(loggerFactory, nameof(loggerFactory));
+            Guard.NotNull(crossChainTransferStore, nameof(crossChainTransferStore));
+            Guard.NotNull(asyncLoopFactory, nameof(asyncLoopFactory));
+            Guard.NotNull(nodeLifetime, nameof(nodeLifetime));
+            Guard.NotNull(federationGatewaySettings, nameof(federationGatewaySettings));
+
+            this.loggerFactory = loggerFactory;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.crossChainTransferStore = crossChainTransferStore;
+            this.asyncLoopFactory = asyncLoopFactory;
+            this.nodeLifetime = nodeLifetime;
+            this.connectionManager = connectionManager;
+            this.federationGatewaySettings = federationGatewaySettings;
+        }
+
+        /// <summary>
+        /// Broadcast the partial transaction request to federation members.
+        /// </summary>
+        /// <param name="payload">The payload to broadcast.</param>
+        async Task BroadcastAsync(RequestPartialTransactionPayload payload)
```

Comment:
Maybe move documentation over to the interface.

Reply:
Done.